File: advent-of-code-2023/dart/test/day03_test.dart
import 'package:test/test.dart';

import '../bin/day03.dart';

void main() {

  var day03 = Day03.using("../input/day03.txt");

  test('day 3 test part 1', () {
    expect(day03.part1(), 537832);
  });

  test('day 3 test part 2', () {
    expect(day03.part2(), 81939900);
  });
}

File: advent-of-code-2024/src/test/kotlin/aoc/historianhysteria/Day03Test.kt
package aoc.historianhysteria

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class Day03Test {

    private companion object {
        private val day03 = Day03("input/day03.txt")
        private val day03Java = Day03Java("input/day03.txt")
        private val day03dr = Day03("input/day03-dr.txt")
    }

    @Test
    fun part1dr() {
        day03dr.part1() shouldBe 153_469_856
    }

    @Test
    fun part2dr() {
        day03dr.part2() shouldBe 77_055_967
    }

    @Test
    fun part1() {
        val expected = 161_085_926
        day03Java.part1() shouldBe expected
        day03.part1() shouldBe expected
    }

    @Test
    fun part2() {
        val expected = 82_045_421
        day03Java.part2() shouldBe expected
        day03.part2() shouldBe expected
    }

}